aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -27,37 +27,192 @@ Kestrel supports the following features:
 
 Kestrel is supported on all platforms and versions that .NET Core supports.
 
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
 [View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/servers/kestrel/sample)
 
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/servers/kestrel/sample2)
+
+---
+
 ## When to use Kestrel with a reverse proxy
 
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
 If your application accepts requests only from an internal network, you can use Kestrel by itself.
 
-![Kestrel to internal network](kestrel/_static/kestrel-to-internal.png)
+![Kestrel communicates directly with your internal network](kestrel/_static/kestrel-to-internal.png)
 
 If you expose your application to the Internet, you must use IIS, Nginx, or Apache as a *reverse proxy server*. A reverse proxy server receives HTTP requests from the Internet and forwards them to Kestrel after some preliminary handling.
 
-![Kestrel to Internet](kestrel/_static/kestrel-to-internet.png)
+![Kestrel communicates indirectly with the Internet through a reverse proxy server, such as IIS, Nginx, or Apache](kestrel/_static/kestrel-to-internet.png)
 
 A reverse proxy is required for edge deployments (exposed to traffic from the Internet) for security reasons. Kestrel is relatively new and does not yet have a full complement of defenses against attacks. This includes but isn't limited to appropriate timeouts, size limits, and concurrent connection limits.
 
-Another scenario that requires a reverse proxy is when you have multiple applications that share the same port running on a single server. That doesn't work with Kestrel directly because Kestrel doesn't support sharing a port between multiple processes. When you configure Kestrel to listen on a port, it handles all traffic for that port regardless of host header. A reverse proxy that can share ports must then forward to Kestrel on a unique port.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+If your application accepts requests only from an internal network, you can use Kestrel by itself.
+
+![Kestrel communicates directly with your internal network](kestrel/_static/kestrel-to-internal.png)
+
+If you expose your application to the Internet, you can use Kestrel by itself or with a *reverse proxy server*, such as IIS, Nginx, or Apache. A reverse proxy server receives HTTP requests from the Internet and forwards them to Kestrel after some preliminary handling.
```

Comment:
I would rewrite this section to not distinguish between internal and edge deployments. Perhaps you could mention that using a reverse proxy can better integrate with existing infrastructure and provide an optional additional layer of configuration and defense.